rust_src/remacs-sys/lib.rs

```diff
@@ -286,6 +297,97 @@ pub struct Lisp_Marker {
     pub bytepos: ptrdiff_t,
 }
 
+/// Represents the cursor position within an Emacs window. For
+/// documentation see stuct cursor_pos in window.h.
+#[repr(C)]
+pub struct CursorPos {
+    // Pixel position.  These are always window relative.
+    x: c_int,
+    y: c_int,
+    // Glyph matrix position.
+    hpos: c_int,
+    vpos: c_int,
+}
+
+/// Represents an Emacs window. For documentation see struct window in
+/// window.h.
+#[repr(C)]
+pub struct Lisp_Window {
+    pub header: Lisp_Vectorlike_Header,
+    pub frame: Lisp_Object,
+    pub next: Lisp_Object,
+    pub prev: Lisp_Object,
+    pub parent: Lisp_Object,
+    pub normal_lines: Lisp_Object,
+    pub normal_cols: Lisp_Object,
+    pub new_total: Lisp_Object,
+    pub new_normal: Lisp_Object,
+    pub new_pixel: Lisp_Object,
+    pub contents: Lisp_Object,
+    pub start: Lisp_Object,
+    pub pointm: Lisp_Object,
+    pub old_pointm: Lisp_Object,
+    pub temslot: Lisp_Object,
+    pub vertical_scroll_bar: Lisp_Object,
+    pub vertical_scroll_bar_type: Lisp_Object,
+    pub horizontal_scroll_bar: Lisp_Object,
+    pub horizontal_scroll_bar_type: Lisp_Object,
+    pub display_table: Lisp_Object,
+    pub dedicated: Lisp_Object,
+    pub redisplay_end_trigger: Lisp_Object,
+    pub combination_limit: Lisp_Object,
+    pub window_paremeters: Lisp_Object,
```

Comment:
Sorry about this. I think I accidentally reverted my first patch with my second patch. I'm fairly new to Github, so I appreciate everyone's patience.